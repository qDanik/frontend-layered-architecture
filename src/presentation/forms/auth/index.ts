import { AppContainer } from 'containers';
import { ServiceIdentifier } from 'containers/config';
import { AuthFormFields, AuthToken } from 'domain/auth';
import { AuthForm, AuthFormSubmitResponse } from './auth.form';
import { BaseForm } from '../base.form';

const AuthFormType: ServiceIdentifier<BaseForm<AuthFormFields, Promise<AuthToken>>> =
  Symbol('AuthForm');

AppContainer.bind(AuthFormType).to(AuthForm);

export { AuthForm, AuthFormType };
export type { AuthFormSubmitResponse };
