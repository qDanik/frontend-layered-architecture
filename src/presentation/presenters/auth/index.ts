import { AppContainer } from 'containers';
import { ServiceIdentifier } from 'containers/config';

import type { AuthPresenter } from './auth.presenter';
import { AuthPresenterImpl } from './auth.presenter.impl';

const AuthPresenterType: ServiceIdentifier<AuthPresenter> = Symbol('AuthPresenter');

AppContainer.bind(AuthPresenterType).to(AuthPresenterImpl);

export { AuthPresenter, AuthPresenterType, AuthPresenterImpl };
