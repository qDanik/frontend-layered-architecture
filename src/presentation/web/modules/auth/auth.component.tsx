import React from 'react';
import { Form, Button, Input } from 'presentation/web/components/form';
import { AuthProps } from './auth.typings';

export function Auth({ presenter }: AuthProps): React.ReactElement {
  return (
    <Form entity={presenter?.form}>
      <Input label={presenter?.t('auth.form.labels.login')} name="login" />
      <Button>{presenter?.t('button:login')}</Button>
    </Form>
  );
}
