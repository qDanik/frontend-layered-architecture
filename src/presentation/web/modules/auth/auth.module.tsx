import React from 'react';

import { Form, FormButton, FormInput } from 'presentation/web/components/form';

import { AuthProps } from './auth.typings';

export function Auth({ presenter }: AuthProps): React.ReactElement {
  return (
    <Form entity={presenter?.form}>
      <FormInput label={presenter?.t('auth.form.labels.login')} name="login" />
      <FormButton>{presenter?.t('button.login')}</FormButton>
    </Form>
  );
}
