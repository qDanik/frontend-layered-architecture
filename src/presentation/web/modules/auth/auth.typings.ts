import { AuthPresenter } from 'presentation/presenters/auth/auth.presenter';

export interface AuthProps {
  presenter: AuthPresenter;
}
