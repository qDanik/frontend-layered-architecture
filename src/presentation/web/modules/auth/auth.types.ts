import { AuthPresenter } from 'presentation/presenters/auth';

export interface AuthProps {
  presenter: AuthPresenter;
}
