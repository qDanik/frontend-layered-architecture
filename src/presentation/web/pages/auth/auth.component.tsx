import React from 'react';
import { AuthPresenterType } from 'presentation/presenters/auth';
import { useInjection } from 'presentation/web/components';
import { Auth } from 'presentation/web/modules/auth';

export function AuthPage(): React.ReactElement {
  const authPresenter = useInjection(AuthPresenterType);

  return <Auth presenter={authPresenter} />;
}
