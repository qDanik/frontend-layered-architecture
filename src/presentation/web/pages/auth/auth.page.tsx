import React from 'react';

import { AuthPresenterType } from 'presentation/presenters/auth';
import { useInjection, withErrorBoundary } from 'presentation/shared';
import { Auth } from 'presentation/web/modules/auth';

function AuthPageComponent(): React.ReactElement {
  const authPresenter = useInjection(AuthPresenterType);

  return <Auth presenter={authPresenter} />;
}

export const AuthPage = withErrorBoundary(AuthPageComponent);
