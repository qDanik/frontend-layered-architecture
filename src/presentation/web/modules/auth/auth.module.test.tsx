import React from 'react';
import { act, render } from '@testing-library/react';

import { Auth } from './auth.module';

describe('auth.module', () => {
  it('should match snapshot', () => {
    const { container } = render(<Auth presenter={null} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
