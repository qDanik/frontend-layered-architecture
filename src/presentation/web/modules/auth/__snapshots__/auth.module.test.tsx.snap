// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth.module should match snapshot 1`] = `
.c3 {
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
  height: 52px;
  width: 100%;
  padding: 4px 5px 0;
  font-size: 20px;
  font-weight: 600;
  text-transform: uppercase;
  border-radius: 52px;
  transition: box-shadow 0.2s;
  overflow: hidden;
  color: gray;
  background-color: lightgray;
}

.c0 {
  display: flex;
  flex-direction: column;
  margin-bottom: 20px;
  width: 100%;
}

.c1 {
  display: block;
  margin-left: 15px;
  margin-bottom: 2px;
  color: var(--gray);
  font-size: 16px;
  font-weight: 600;
}

.c2 {
  position: relative;
  display: block;
  max-height: 70px;
  padding: 15px;
  margin: 2px 0;
  color: black;
  font-size: 16px;
  font-weight: 600;
  border: none;
  border-radius: 15px;
  background-color: lightgray;
  border: 1px solid transparent;
  transition: border 0.2s;
  outline: none;
}

.c2:focus {
  border: 1px solid orange!important;
}

.c2::-webkit-input-placeholder {
  color: gray;
  font-weight: 500;
}

<form>
  <logosvg
    style="display: block; height: 100px; margin: 10px auto; width: 100px;"
  />
  <label
    class="c0"
  >
    <span
      class="c1"
    />
    <input
      class="c2"
      name="login"
      value=""
    />
  </label>
  <button
    class="c3"
    disabled=""
  />
</form>
`;
